File: Sponsors/mw_streams_python/mws/pipeline.py
from mws.operators.operator import Operator, Source, Sink


class Pipeline:
    def __init__(self, source: Operator) -> None:
        self.source = source
        self.head = source

    @staticmethod
    def of(source: Source) -> "Pipeline":
        return Pipeline(source=source)

    def then(self, operator: Operator) -> "Pipeline":
        self.head.add_child(operator)
        self.head = operator

        return self

    def to(self, sink: Sink) -> None:
        self.head.add_child(sink)
        self.head = sink

        return self

    def start(self) -> None:
        self.source.start()

File: Sponsors/mw-streams-java/src/main/java/com/mwam/hackathon/pipeline/Pipeline.java
package com.mwam.hackathon.pipeline;

import com.mwam.hackathon.sinks.Sink;
import com.mwam.hackathon.sources.Source;

/**
 * @param <I>
 * @param <O>
 */
public class Pipeline<I, O> {
    private final Source<I> source;
    private final Sink<O> sink;


    public Pipeline(Source<I> source, Sink<O> sink) {
        this.source = source;
        this.sink = sink;
    }

    public static <I> SourcePipeline<I> from(Source<I> source) {
        return new SourcePipeline<>(source);
    }

    public void start() {
        this.source.start();
    }
}


